src/test/java/com/flightstats/hub/app/GuiceToHK2AdapterTest.java

```diff
@@ -0,0 +1,154 @@
+package com.flightstats.hub.app;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Provides;
+import com.google.inject.name.Named;
+import lombok.Value;
+import org.glassfish.hk2.api.ServiceLocator;
+import org.glassfish.hk2.api.ServiceLocatorFactory;
+import org.glassfish.hk2.utilities.ServiceLocatorUtilities;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class GuiceToHK2AdapterTest {
+
+    @Value
+    private static class SimpleService {
+        String name;
+
+        SimpleService() {
+            this("unnamed");
+        }
+
+        SimpleService(String name) {
+            this.name = name;
+        }
+
+    }
+
+    private static class NativeService {
+        @javax.inject.Inject
+        NativeService() {}
+    }
+
+    private static class GuicedService {
+        @com.google.inject.Inject
+        GuicedService() {}
+    }
+
+    @javax.inject.Singleton
+    private static class NativeSingletonService {}
+
+    @com.google.inject.Singleton
+    private static class GuicedSingletonService {}
+
+    @Test
+    public void testSimple() {
+        ServiceLocator locator = initializeHK2(buildEmptyInjector());
+        locator.getService(NativeService.class);
+        locator.getService(GuicedService.class);
+    }
+
+    @Test
+    public void testProvides() {
+        Injector injector = Guice.createInjector(new AbstractModule() {
+            @Override
+            protected void configure() {}
+
+            @Provides
+            SimpleService providesService() {
+                return new SimpleService();
+            }
+        });
+        ServiceLocator locator = initializeHK2(injector);
+        locator.getService(SimpleService.class);
```

Comment:
I'd probably add an assertion against `getName()` or that the service isn't null or the like, just to have an assertion here...

Reply:
or possibly that if you initialize two services, you get different instances? I dunno.

Anyway, thanks for humoring my questions about how singleton stuff works. Looks good to me!

Reply:
I had convinced myself that relying on `getService` to throw an exception was good enough but after further reading I should be checking for null.

From the JavaDoc:
>@return An instance of the contract or impl.  May return null if there is no provider that provides the given implementation or contract
> @throws MultiException if there was an error during service creation